### PR TITLE
fix: remove GOFIPS from build as this is runtime check

### PR DESCRIPTION
**Pull request by sumo-drosiek**

### CODE REVIEW

**Comment by ccressent** [comment]
Do you think that will address #1387?
I'm curious because it seems to me like the issue is also about the `requirefips` build flag?

**Comment by swiatekm-sumo** [comment]
Does GOFIPS even do anything at build time?

**Comment by ccressent** [comment]
I don't believe so, I think this is a runtime configuration only for the Microsoft version of the Go toolchain: https://github.com/microsoft/go/blob/microsoft/main/eng/doc/fips/README.md#usage-runtime

**Comment by ccressent** [comment]
I think this PR can be abandoned?
It was opened while @sumo-drosiek was looking at #1387, but this has been handled by other means.

**Comment by swiatekm-sumo** [comment]
@sumo-drosiek can you just delete GOFIPS from the Makefile, if it doesn't do anything?

File: otelcolbuilder/Makefile [hunk 1]
```diff
@@ -50,9 +50,9 @@ endif
 ifeq ($(FIPS_SUFFIX),"")
 GOEXPERIMENT=""
 LDFLAGS="-s -w"
-GOFIPS=1
 # We want to use boringcrypto for fips setup
 else
+GOFIPS=1
```
Comment (sumo-drosiek) [inline otelcolbuilder/Makefile:55]:
```suggestion
```

File: otelcolbuilder/Makefile [hunk 2]
```diff
@@ -53,9 +53,7 @@ endif
 # We don't want to strip symbols for FIPS builds, as they're useful for verifying the build
 ifeq ($(FIPS_SUFFIX),"")
 GOEXPERIMENT=""
-LDFLAGS+= -s -w
-GOFIPS=1
-
+LDFLAGS="-s -w"
```
Comment (sumo-drosiek) [inline otelcolbuilder/Makefile:56]:
```suggestion
LDFLAGS+= -s -w
```

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**